doc/plugin_tutorial/tuto1/src/simple_declare.ml

```diff
@@ -1,8 +1,6 @@
-let edeclare ?hook ~name ~poly ~scope ~kind ~opaque ~udecl ~impargs sigma body tyopt  =
-  DeclareDef.declare_definition ~name ~scope ~kind ~impargs ?hook
-    ~opaque ~poly ~udecl ~types:tyopt ~body sigma
-
 let declare_definition ~poly name sigma body =
   let udecl = UState.default_univ_decl in
-  edeclare ~name ~poly ~scope:(DeclareDef.Global Declare.ImportDefaultBehavior)
-    ~kind:Decls.(IsDefinition Definition) ~opaque:false ~impargs:[] ~udecl sigma body None
+  let scope = DeclareDef.Global Declare.ImportDefaultBehavior in
+  let kind = Decls.(IsDefinition Definition) in
+  DeclareDef.declare_definition ~name ~scope ~kind ~impargs:[] ~udecl
```

Comment:
Not directly related to this PR, but this kind of use shows that `impargs` should probably be an optional parameter.

Reply:
About the status of impargs, I have #11272 which makes significant simplifications of the implicit arguments paths. It only depends on #11099 and #11135 now, so, its turn is soon coming.

Reply:
It is quite hard to know when to turn a parameters into implicits; I understand large OCaml codebases actually consider implicits as evil as experimental evidence shows people just forget to use them when they are needed. I subscribe to this view.

In the particular case of `Declare` my plan is to use an `Info.t` record in the way it is done in Lemmas ; that IMHO provides the best of both approaches.